central/graphql/resolvers/plotted_node_vulnerabilities.go

```diff
@@ -48,15 +85,23 @@ func (pvr *PlottedNodeVulnerabilitiesResolver) BasicNodeVulnerabilityCounter(_ c
 }
 
 // NodeVulnerabilities returns the node vulnerabilities for top risky nodes scatter-plot
-func (pvr *PlottedNodeVulnerabilitiesResolver) NodeVulnerabilities(ctx context.Context, args PaginatedQuery) ([]NodeVulnerabilityResolver, error) {
-	vulnResolvers, err := unwrappedPlottedVulnerabilities(ctx, pvr.root, pvr.all, args)
-	if err != nil {
-		return nil, err
+func (pvr *PlottedNodeVulnerabilitiesResolver) NodeVulnerabilities(ctx context.Context, args PaginationWrapper) ([]NodeVulnerabilityResolver, error) {
+	if !features.PostgresDatastore.Enabled() {
+		vulnResolvers, err := unwrappedPlottedVulnerabilities(ctx, pvr.root, pvr.all, PaginatedQuery{Pagination: args.Pagination})
+		if err != nil {
+			return nil, err
+		}
+
+		ret := make([]NodeVulnerabilityResolver, 0, len(vulnResolvers))
+		for _, resolver := range vulnResolvers {
+			ret = append(ret, resolver)
+		}
+		return ret, nil
 	}
 
-	ret := make([]NodeVulnerabilityResolver, 0, len(vulnResolvers))
-	for _, resolver := range vulnResolvers {
-		ret = append(ret, resolver)
+	if len(pvr.all) == 0 {
+		return nil, nil
 	}
-	return ret, nil
+	q := search.NewQueryBuilder().AddExactMatches(search.CVEID, pvr.all...).Query()
+	return pvr.root.NodeVulnerabilities(ctx, PaginatedQuery{Query: &q, Pagination: args.Pagination})
```

Comment:
you can remove the postgres check from here since the top-level node vulnerabilities resolver handles the fork.

```
	q := search.NewQueryBuilder().AddExactMatches(search.CVEID, pvr.all...).Query()
	return pvr.root.NodeVulnerabilities(ctx, PaginatedQuery{Query: &q, Pagination: args.Pagination})
```